verifiers/utils/builder_test.go

```diff
@@ -201,6 +202,85 @@ func Test_Matches(t *testing.T) {
 			match:     "some/name@vla@blo",
 			err:       serrors.ErrorInvalidFormat,
 		},
+		// Same as above with `allowRef: true`.
+		{
+			name:      "match full",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name@v1.2.3",
+			allowRef:  true,
+		},
+		{
+			name:      "match name",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name",
+			allowRef:  true,
+		},
+		{
+			name:      "mismatch name",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name2",
+			allowRef:  true,
+			err:       serrors.ErrorMismatchBuilderID,
+		},
+		{
+			name:      "mismatch version",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name@v1.2.4",
+			allowRef:  true,
+			err:       serrors.ErrorMismatchBuilderID,
+		},
+		{
+			name:      "invalid empty version",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name@",
+			allowRef:  true,
+			err:       serrors.ErrorInvalidFormat,
+		},
+		{
+			name:      "too many '@' - need version",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name@vla@blo",
+			allowRef:  true,
+			err:       serrors.ErrorInvalidFormat,
+		},
+		{
+			name:      "too many '@' - no need version",
+			builderID: "some/name@v1.2.3",
+			match:     "some/name@vla@blo",
+			allowRef:  true,
+			err:       serrors.ErrorInvalidFormat,
+		},
+		// Mismatch of tag length.
+		{
+			name:      "match long tag match short",
+			builderID: "some/name@refs/tags/v1.2.3",
+			match:     "some/name@v1.2.3",
+			allowRef:  true,
+		},
+		{
+			name:      "long tag match short no ref",
+			builderID: "some/name@refs/tags/v1.2.3",
+			match:     "some/name@v1.2.3",
+			err:       serrors.ErrorMismatchBuilderID,
+		},
+		{
+			name:      "match long tags",
+			builderID: "some/name@refs/tags/v1.2.3",
+			match:     "some/name@refs/tags/v1.2.3",
+			allowRef:  true,
+		},
+		{
+			name:      "mismatch tag length",
+			builderID: "some/name@refs/tags/v1.2.3",
+			match:     "some/name@v1.2.3",
+			err:       serrors.ErrorMismatchBuilderID,
```

Comment:
sorry if I'm a little lost here, but does a user provide `match` or `builderID`? I thought they provide `match`. Shouldn't it allow user to pass with the short name to provide?

Reply:
user provides `match`. BuilderID is the one in the provenance. Let me make this clearer in the doc

Reply:
> Shouldn't it allow user to pass with the short name to provide?

You're correct, but only if `allowRef: true`, which is not set here. The `allowRef` is to ensure we nly enable this feature for GHA builders. The GCB builder just use `name@version`

Reply:
I renamed to TrustedBuilderID, to make to clear this is the trusted one from the cert / provenance.

Reply:
np!